internal/renderers/skia/itemrenderer.rs

```diff
@@ -182,24 +182,23 @@ impl<'a> SkiaItemRenderer<'a> {
 
         self.canvas.save();
 
-        let mut source_rect = item.source_clip().map_or_else(
-            || skia_safe::Rect::from_wh(skia_image.width() as _, skia_image.height() as _),
-            |r| {
-                skia_safe::Rect::from_xywh(
-                    r.min_x() as _,
-                    r.min_y() as _,
-                    r.width() as _,
-                    r.height() as _,
-                )
-            },
+        let fit = i_slint_core::graphics::fit(
+            item.image_fit(),
+            dest_rect.size,
+            item.source_clip()
+                .unwrap_or_else(|| euclid::rect(0, 0, skia_image.width(), skia_image.height())),
+            self.scale_factor,
         );
-        adjust_to_image_fit(item.image_fit(), &mut source_rect, &mut dest_rect);
 
-        self.canvas.clip_rect(to_skia_rect(&dest_rect), None, None);
+        let dst = to_skia_rect(&PhysicalRect::new(fit.offset, fit.size));
+        self.canvas.clip_rect(dst, None, None);
 
-        let transform =
-            skia_safe::Matrix::rect_to_rect(source_rect, to_skia_rect(&dest_rect), None)
-                .unwrap_or_default();
+        let transform = skia_safe::Matrix::rect_to_rect(
+            to_skia_rect(&PhysicalRect::from_untyped(&fit.clip_rect.cast())),
```

Comment:
Since clip_rect is in source coordinates, wouldn't it be a little cleaner to go straight to skia_safe::Rect::from_xywh() or so, instead of the intermediate PhysicalRect?

(Not too strong of a feeling here, just trying to avoid from_untyped)